application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -65,26 +67,34 @@
         $response.setStatus(403)
       #end
     #end
-  #elseif("$!request.action" == "addtopic")
+  #elseif("$!request.action" == 'addtopic')
     #if("$!request.title.trim()" != '')
       #if("$!request.space" != '')
-        #set($newTopicName = $request.page)
```

Comment:
Note that this page name was auto-generated to avoid collisions. Using, in the NS case, just the topic title means that we can have collisions.

This could be handled by emulating the behavior of `xwiki.getUniquePageName()` but for space names:
* You could use a while loop and increment a counter to append to a space name and check if it already exists. See how [`xwiki.getUniquePageName()` is implemented](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java#L6186)
* Additionally, if you need a random string, you can use [`$util.generateRandomString(int)`](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Util.java#L129).

If you want to preserve the same type of space names (i.e. Topic_0, Topic_1, etc, go with the first option).